source/code/plugin/in_kube_logs.rb

```diff
@@ -0,0 +1,199 @@
+#!/usr/local/bin/ruby
+
+module Fluent
+    
+    class Kube_Logs_Input < Input
+        Plugin.register_input('kubelogs', self)
+    
+        @@KubeLogsStateFile = "/var/opt/microsoft/docker-cimprov/state/KubeLogQueryState.yaml"
+    
+        def initialize
+            super
+            require 'yaml'
+            require 'date'
+            require 'time'
+            require 'json'
+
+            require_relative 'KubernetesApiClient'
+            require_relative 'oms_common'
+            require_relative 'omslog'  
+        end
+    
+        config_param :run_interval, :time, :default => '1m'
+        config_param :tag, :string, :default => "oms.api.KubeLogs"
+
+        def configure (conf)
+            super
+        end
+    
+        def start
+            if KubernetesApiClient.isValidRunningNode && @run_interval
+                @finished = false
+                @condition = ConditionVariable.new
+                @mutex = Mutex.new
+                @thread = Thread.new(&method(:run_periodic))
+            else
+                enumerate
+            end
+        end
+    
+        def shutdown
+            if KubernetesApiClient.isValidRunningNode && @run_interval
+                @mutex.synchronize {
+                    @finished = true
+                    @condition.signal
+                }
+                @thread.join
+            end  
+        end
+    
+        def enumerate(podList = nil)
+
+            namespace = ENV['OMS_KUBERNETES_LOGS_NAMESPACE']
+            if namespace.nil? || namespace.empty? 
+                return
+            end
+
+            time = Time.now.to_f
+            if KubernetesApiClient.isValidRunningNode
+
+                if podList.nil?
+                    pods = KubernetesApiClient.getPods(namespace)
+                else
+                    pods = podList
+                end   
+                logQueryState = getLogQueryState
+                newLogQueryState = {}
+
+                pods.each do |pod|
+                    record = {}
+                    begin
+                        pod['status']['containerStatuses'].each do |container|
+        
+                            # if container['state']['running']
+                            #     puts container['name'] + ' is running'
+                            # end
+
+                            timeStamp = DateTime.now
+
+                            containerId = pod['metadata']['namespace'] + "_" + pod['metadata']['name'] + "_" + container['name']
+                            if !logQueryState.empty? && logQueryState[containerId]
+                                timeStamp = DateTime.parse(logQueryState[containerId])
+                            end  
+
+                            # Try to get logs for the container
+                            begin
+                              $log.info "Getting logs for #{container['name']}"
```

Comment:
Do you still need these logs for troubleshooting ? Perhaps, you might consider setting them to debug level if you dont need them for now.